01-html/01-koajs.html

```diff
@@ -1,4 +1,671 @@
+      <p>This includes methods for common tasks like content-negotiation, cache freshness, proxy support, and redirection among others. Despite supplying a reasonably large number of helpful methods Koa maintains a small footprint, as no middleware are bundled.</p>
+      <p>The obligatory hello world application:</p>
+        <code>var koa = require('koa');<br>var app = koa();
+          <br>app.use(function *(){
+          <br>this.body = 'Hello World';
+          <br>});
+          <br>app.listen(3000);</code>
+      <h3>Cascading</h3>
+        <p>Koa middleware cascade in a more traditional way as you may be used to with similar tools - this was previously difficult to make user friendly with node's use of callbacks. However with generators we can achieve <q>true</q> middleware. Contrasting Connect's implementation which simply passes control through series of functions until one returns, Koa yields <q>downstream</q>, then control flows back <q>upstream</q>.</p>
+        <p>The following example responds with <q>Hello World</q>, however first the request flows through the <em>x-response-time</em> and <em>logging</em> middleware to mark when the request started, then continue to yield control through the response middleware. When a middleware invokes <em>yield next</em> the function suspends and passes control to the next middleware defined. After there are no more middleware to execute downstream, the stack will unwind and each middleware is resumed to perform its upstream behaviour.</p>
+        <code>var koa = require('koa');
+          <br>var app = koa();
+        </code>
+        <code>// x-response-time</code>
+        <code>app.use(function *(next){
+          <br>var start = new Date;
+          <br>yield next;
+          <br>var ms = new Date - start;
+          <br>this.set('X-Response-Time', ms + 'ms';)
+          <br>});
+        </code>
+        <code>//logger</code>
+        <code>app.use(function *(next) {
+          <br>var start = new Date;
+          <br>yield next;
+          <br>var ms = newDate - start;
+          <br>console.log('%s %s - %s', this.method, thi.url, ms);
+          <br>});
+        </code>
+        <code>//response</code>
+        <code>app.use(function *(){
+          <br>this.body = 'Hello World';
+          <br>});
+        </code>
+        <code>app.listen(3000);</code>
+      <h3>Settings</h3>
+        <p>Application settings are properties on the <code>app</code> instance,
+          currently the following are supported:</p>
+          <ul>
+            <li><em>app.name</em> optionally give your application a name</li>
+            <li><em>app.env</em> defaulting to the <b>NODE_ENV</b> or <q>development</q></li>
```

Comment:
`<strong>` is generally preferred over `<b>` these days. Why? Don't ask questions. IT IS JUST BETTER, OK?